### ISSUE TITLE
PurgeCSS Vite Plugin

### ISSUE THREAD

**Issue by endigo9740** [skeletonlabs/skeleton#2881]
### Link to the Page

- v2 Docs: https://www.skeleton.dev/docs/purgecss
- GitHub: https://github.com/AdrianGonz97/vite-plugin-tailwind-purgecss

### Describe the Issue (screenshots encouraged!)

We need to verify this plugin is working with Skeleton v3. It can help dramatically decrease bundle sizes by purging unwanted CSS in the bundle. However there is some uncertainty on whether it will work, especially with the Vite 6 release recently

## How to Test

For the initial test we'll ONLY test SvelteKit as we know it was working before in v2. Do the following:

1. Create a new Skeleton v3 project for SvelteKit
2. Run the `npm run build` command and view the size of the CSS bundle. It should be listed in the console during this process.
3. Then, install the PurgeCSS plugin per the instructions on the docs linked above. I'll also share the readme below
4. Build again, and compare the size of the CSS bundle. If it's working this should be reduced.

## Handling the Results

There's two outcomes here:

1. If the size doesn't go down, then we know it's not working and can aim to update the plugin asap
2. If it is working, then we'll need to port that v2 doc over to the v3 docs under the Guides section

## Follow Up Test

If it works in SvelteKit, ensure we've tested it in all Vite-based frameworks, including:

- [SvelteKit](https://next.skeleton.dev/docs/get-started/installation/sveltekit)
- [Vite/Svelte](https://next.skeleton.dev/docs/get-started/installation/vite-svelte)
- [Vite/React](https://next.skeleton.dev/docs/get-started/installation/vite-react)
- [Astro](https://next.skeleton.dev/docs/get-started/installation/astro)

Note that Next.js uses Webpack/Turbopack, not Vite, and will not qualify for this test.

## Reporting the Results

When tests are complete, ping me here (Chris) as well as @AdrianGonz97 (CokaKoala on Discord) to share the results. Adrian is the original author and can address any issues that might arise.

**Comment by phamduylong** [comment]
@endigo9740 anything blocking this one? I can give this a go if it's just porting the same markdown over, into an mdx file

**Comment by endigo9740** [comment]
@phamduylong yeah @AdrianGonz97 is the author of the plugin and was going to confirm it works for all supported meta-frameworks listed above. This may need additional effort now that Vite 6 has been released. So we'll need his thumbs up before we can progress on this.

**Comment by endigo9740** [comment]
![Image](https://github.com/user-attachments/assets/59b368e8-9d59-40bb-8bda-7c80a3cdc2c2)

**Comment by nullpointerexceptionkek** [comment]
I will update this comment along the way.

The peer deps for tailwind should be updated to 4.0.0

**Comment by endigo9740** [comment]
@AdrianGonz97 just pinging you here to make sure you're aware. This is great news though!
